### DISCUSSION TITLE
自定义函数  计算DECIMAL 入库时会自动 四舍五入 为什么

### DISCUSSION THREAD

**Discussion by foundation1** [apache/doris#32443] [A - General / Q&A]
下面是我的 插入数据  sql 
`INSERT INTO jvs_data_factory_ods.ods_fae2d548129be8c110bf8f7273d70f05_7175686911355260928_false ( `950718913513951232`, `6c8001b841f871978d0798586b98ed0a`, `2a17bcd2f28fa06cc115799d0449f5a2`, `1710817080780` ) SELECT
`950718913513951232`,
`6c8001b841f871978d0798586b98ed0a`,
`2a17bcd2f28fa06cc115799d0449f5a2`,
JVS_DIVIDE ( `2a17bcd2f28fa06cc115799d0449f5a2`, `6c8001b841f871978d0798586b98ed0a` ) 
FROM
	jvs_data_factory_ods.ods_fae2d548129be8c110bf8f7273d70f05_7175686717855240192_false`
建表语句如下
`CREATE TABLE IF NOT EXISTS jvs_data_factory_ods.ods_fae2d548129be8c110bf8f7273d70f05_7175686911355260928_false(
`950718913513951232` STRING COMMENT "知识库id",
`6c8001b841f871978d0798586b98ed0a` DECIMAL(18,0) COMMENT "排序字段",
`2a17bcd2f28fa06cc115799d0449f5a2` DECIMAL(18,0) COMMENT "文档大小",
`1710817080780` DECIMAL(18,5) COMMENT "cs",
`sys_ods_bucket` DATETIME DEFAULT CURRENT_TIMESTAMP COMMENT "插入时间-用于分桶策略" 
)DISTRIBUTED BY HASH( `sys_ods_bucket`) BUCKETS AUTO 
PROPERTIES (
"enable_duplicate_without_keys_by_default" = "true","replication_num" = "2"
);`  
其中 1710817080780 字段的类型为 DECIMAL(18,5)
执行上面  插入sql中的select语句  结果如下
![image](https://github.com/apache/doris/assets/28500506/8ff15d1e-b375-4862-a045-55af18d15812)

实际值为 1.82417  但是入库后的结果为 2.0
![image](https://github.com/apache/doris/assets/28500506/c0ab5f6e-dc95-412c-b090-47467b1c16cf)

这个是为什么  其中JVS_DIVIDE  是我的自定义函数  就是一个除法 代码如下
![image](https://github.com/apache/doris/assets/28500506/2c6dd708-8f53-4ca7-b2c9-4e38698e9951)

**Comment by foundation1** [comment]
而且我如果使用 doris  的除法  sql如下  `INSERT INTO jvs_data_factory_ods.ods_fae2d548129be8c110bf8f7273d70f05_7175686911355260928_false ( `950718913513951232`, `6c8001b841f871978d0798586b98ed0a`, `2a17bcd2f28fa06cc115799d0449f5a2`, `1710817080780` ) SELECT
`950718913513951232`,
`6c8001b841f871978d0798586b98ed0a`,
`2a17bcd2f28fa06cc115799d0449f5a2`,
`2a17bcd2f28fa06cc115799d0449f5a2`/ `6c8001b841f871978d0798586b98ed0a` 
FROM
	jvs_data_factory_ods.ods_fae2d548129be8c110bf8f7273d70f05_7175686717855240192_false`   插入就是正常的

**Comment by yagagagaga** [comment]
Maybe this function [round](https://doris.apache.org/zh-CN/docs/dev/sql-manual/sql-functions/numeric-functions/round) can help you.
